Assets/Source/Components/UI/HealthBarComponent.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Assets.Source.Components.Actor;
+using Assets.Source.Components.Base;
+using Assets.Source.Constants;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace Assets.Source.Components.UI
+{
+    public class HealthBarComponent : ComponentBase
+    {
+        private Animator animator;
+        private ActorBehavior actorBehavior;
+        private Image healthImage;
+        private int presentedHealth;
+
+        public override void ComponentStart()
+        {
+            animator = GetRequiredComponent<Animator>();
+            healthImage = GetRequiredComponent<Image>();
+            GameObject player = GetRequiredObject(GameObjects.Actors.Player);
+            actorBehavior = GetRequiredComponent<ActorBehavior>(player);
+
+            presentedHealth = actorBehavior.Health;
+
+            base.ComponentStart();
+        }
+
+        public override void ComponentUpdate()
+        {
+            if(actorBehavior.Health < presentedHealth)
+            {
+                animator.SetBool("highlight", true);
+                presentedHealth--;
```

Comment:
What if the health is greater than the presented health?  Like when we add healing items or something

Reply:
this is MUCH better though, having the logic work like this <3

Reply:
I agree